platforms/core-runtime/internal-instrumentation-api/src/main/java/org/gradle/internal/instrumentation/api/jvmbytecode/JvmBytecodeCallInterceptor.java

```diff
@@ -16,18 +16,19 @@
 
 package org.gradle.internal.instrumentation.api.jvmbytecode;
 
-import org.gradle.api.NonNullApi;
+import org.gradle.internal.instrumentation.api.metadata.InstrumentationMetadata;
+import org.gradle.internal.instrumentation.api.types.BytecodeInterceptorFilter;
 import org.gradle.internal.instrumentation.api.types.FilterableBytecodeInterceptor;
 import org.gradle.internal.instrumentation.api.types.FilterableBytecodeInterceptorFactory;
-import org.gradle.internal.instrumentation.api.types.BytecodeInterceptorFilter;
-import org.gradle.internal.instrumentation.api.metadata.InstrumentationMetadata;
-import org.objectweb.asm.MethodVisitor;
+import org.gradle.model.internal.asm.MethodVisitorScope;
 import org.objectweb.asm.tree.MethodNode;
 
+import javax.annotation.Nullable;
 import java.util.function.Supplier;
 
 public interface JvmBytecodeCallInterceptor extends FilterableBytecodeInterceptor {
     boolean visitMethodInsn(
+            MethodVisitorScope mv,
```

Comment:
💭 I changed to pass the visitor from outside to avoid having it available in `findBridgeMethodBuilder` which shouldn't touch bytecode of the method being processed.